### PR TITLE
Enable Bzlmod

### PR DESCRIPTION
**Pull request by lalten**
Incomplete, work in progress :construction:

### CODE REVIEW

File: ros2/interfaces.bzl [hunk 1]
```diff
@@ -821,8 +824,8 @@ py_generator_aspect = aspect(
         ),
         "_py_ext_c_deps": attr.label_list(
             default = [
-                Label("@rules_ros2_python//:python_headers"),
-                Label("@rules_ros2_pip_deps_numpy//:headers"),
+                Label("@rules_python//python/cc:current_py_cc_headers"),
+                Label("@rules_ros2_pip_deps" + ("_311" if BZLMOD_ENABLED else "") + "_numpy//:headers"),
```
Comment (mvukov) [inline ros2/interfaces.bzl:826]: This looks quite ugly IMO. We don't want to constrain folks which Python version they use as long as it's compatible with ROS2 packages. Are you sure that `@rules_ros2_pip_deps_numpy//:headers` doesn't work any more?

**Comment by mvukov** [comment]
Something to keep an eye on: https://github.com/nelhage/rules_boost/pull/522

File: ros2/interfaces.bzl (hunk 1 above)
Comment (lalten) [reply ros2/interfaces.bzl:826]: Agree that's not great... Need to think about how to solve this. A bigger issue at the moment is how to make all the ros2_ http_archive repos work, currently they're not visible from a root module and I'm not sure about the best way to make that work. Can just add a MODULE.bazel to the examples to see it fail.

Comment (mvukov) [reply ros2/interfaces.bzl:826]: I think you can do with https://bazel.build/external/migration#fetch-deps-module-extensions: wrap the autogenerated ros2 http_archives in a function, call that function through a bzlmod extension. In the module file, you'll need to use the `use_repo` for each ros2 repo (tedious at first, I know). Lemme know if you need help.

**Pull request closed without merging**